goal_src/engine/ui/hud-classes.gc

```diff
@@ -7,10 +7,3015 @@
+    (sp-flt spt-vel-y (meters 0.0))
+    (sp-flt spt-vel-z (meters 0.0))
+    (sp-rnd-int-flt spt-rotvel-z (degrees -0.4) 1 145.63556)
+    (sp-int spt-timer -1)
+    (sp-cpuinfo-flags bit3 bit7 bit9)
+    (sp-end)
+    )
+   )
+  )
+
+(set!
+  (-> *part-id-table* 306)
+  (new 'static 'sparticle-launcher
+   :init-specs
+   (new 'static 'inline-array sp-field-init-spec 16
+    (sp-tex spt-texture (new 'static 'texture-id :index #xf :page #x2))
+    (sp-flt spt-num 1.0)
+    (sp-rnd-flt spt-y (meters 0.0) (meters 16.0) 1.0)
+    (sp-flt spt-z 491.52)
+    (sp-flt spt-scale-x (meters 0.4))
+    (sp-copy-from-other spt-scale-y -4)
+    (sp-flt spt-r 128.0)
+    (sp-flt spt-g 255.0)
+    (sp-flt spt-a 24.0)
+    (sp-rnd-flt spt-omega 0.0 65536.0 1.0)
+    (sp-rnd-flt spt-vel-x (meters 0.053333335) (meters 0.053333335) 1.0)
+    (sp-flt spt-vel-y (meters 0.0))
+    (sp-flt spt-vel-z (meters 0.0))
+    (sp-int spt-timer -1)
+    (sp-cpuinfo-flags bit2 bit3 bit7 bit9)
+    (sp-end)
+    )
+   )
+  )
+
+(set!
+  (-> *part-id-table* 307)
+  (new 'static 'sparticle-launcher
+   :init-specs
+   (new 'static 'inline-array sp-field-init-spec 17
+    (sp-tex spt-texture (new 'static 'texture-id :index #xf :page #x2))
+    (sp-flt spt-num 0.25)
+    (sp-flt spt-scale-x (meters 0.35))
+    (sp-copy-from-other spt-scale-y -4)
+    (sp-flt spt-r 255.0)
+    (sp-flt spt-g 255.0)
+    (sp-flt spt-b 0.0)
+    (sp-flt spt-a 96.0)
+    (sp-flt spt-scalevel-x (meters -0.0012500001))
+    (sp-copy-from-other spt-scalevel-y -4)
+    (sp-flt spt-fade-r -2.125)
+    (sp-flt spt-accel-y -0.08192)
+    (sp-int-plain-rnd spt-timer 30 239 1)
+    (sp-cpuinfo-flags bit2 bit9)
+    (sp-int spt-next-time 120)
+    (sp-launcher-by-id spt-next-launcher 308)
+    (sp-end)
+    )
+   )
+  )
+
+(set!
+  (-> *part-id-table* 308)
+  (new 'static 'sparticle-launcher
+   :init-specs
+   (new 'static 'inline-array sp-field-init-spec 2
+    (sp-flt spt-fade-r 0.0)
+    (sp-end)
+    )
+   )
+  )
+
+(deftype hud-pickups (hud)
+  ()
+  :heap-base #xb0
+  :method-count-assert 27
+  :size-assert         #x118
+  :flag-assert         #x1b00b00118
+  )
+
+
+(defmethod dummy-15 hud-pickups ((obj hud-pickups))
+  (let ((t9-0 (method-of-type hud dummy-15)))
+   (t9-0 obj)
+   )
+  (let* ((s5-0 (-> *display* frames (-> *display* on-screen) frame global-buf))
+         (gp-0 (-> s5-0 base))
+         )
+   (let ((s4-0 draw-string-xy))
+    (format (clear *temp-string*) "~D" (-> obj value))
+    (s4-0
+     *temp-string*
+     s5-0
+     (+ (-> obj text-x) (* (-> obj x-sgn) (-> obj offset)))
+     (/
+      (*
+       (+ (-> obj text-y) (* (-> obj y-sgn) (-> obj offset)) (-> obj y-offset))
+       (the int (-> *video-parms* relative-y-scale))
+       )
+      2
+      )
+     (font-color bright-green)
+     (font-flags shadow kerning large)
+     )
+    )
+   (let ((a3-7 (-> s5-0 base)))
+    (let ((v1-8 (the-as object (-> s5-0 base))))
+     (set! (-> (the-as (pointer int64) v1-8)) #x20000000)
```

Comment:
you could cast `v1-8` to a `dma-packet` for this store as well, I think it would look nicer